test-utils/cli/src/lib.rs

```diff
@@ -220,6 +223,25 @@ pub async fn run_node_for_a_while(base_path: &Path, args: &[&str]) {
 	.await
 }
 
+pub async fn block_hash(block_number: u64, url: &str) -> Result<Hash, String> {
+	use substrate_rpc_client::{ws_client, ChainApi};
+
+	let rpc = ws_client(url).await.unwrap();
+
+	let result = ChainApi::<(), Hash, Header, ()>::block_hash(
+		&rpc,
+		Some(ListOrValue::Value(NumberOrHex::Number(block_number.saturating_sub(1)))),
+	)
```

Comment:
This call gives me the block hash of the block that is not corresponding to the `block_number` passed as an argument. It returns the block hash that is associated with the `block_number+1` block.
(That is why I added `saturating_sub(1)` here)

Why is it so, or am I doing something wrong?

Reply:
Do you have some logs you could provide showing this behavior? 

Could you check if you get the same behavior using `curl` and the `chain_getHead` rpc call directly from the command line? I just checked and it seems to be working correctly for me.

<img width="1495" alt="Screenshot 2023-06-20 at 09 46 02" src="https://github.com/paritytech/substrate/assets/16665596/0b2309fd-6928-4ed2-8d81-5dcdf60c15dd">